### PR TITLE
Added acknowledgePurchase feature to BillingManagerV5

### PR DESCRIPTION
**Pull request by mtilhan**
- Added **acknowledgePurchase** feature to BillingManagerV5.

The current removePurchaseFromQueue function can only consume, which leads to incorrect usage for non-consumable products on Google Play. For example, for a "remove-ads" feature that can be purchased once, the user must not purchase it again. Since it is not consumed with this development, it cannot be purchased again.

- Added null check for subscriptionIds in productDetails function. 

If you call getProductsInfo function without subscription id array its stuck and never get success event. Added null check for users dont have subscription on their apps.

### CODE REVIEW

File: actionscript/src/com/freshplanet/ane/AirInAppPurchase/InAppPurchase.as [hunk 1]
```diff
@@ -110,6 +110,19 @@ package com.freshplanet.ane.AirInAppPurchase {
          * @param productId
          * @param receipt
          */
+        public function acknowledgePurchase(productId:String, receipt:String):void {
+
+            if (!isSupported)
+                _dispatchEvent(InAppPurchaseEvent.CONSUME_ERROR, "InAppPurchase not supported");
+            else {
+                if (!_isIOSOrMacOS()) {
```
Comment (dmytrooverchenko) [inline actionscript/src/com/freshplanet/ane/AirInAppPurchase/InAppPurchase.as]: I think it is better to use use `_isAndroid()` here?

File: android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/ExtensionContext.java [hunk 2]
```diff
@@ -158,11 +162,22 @@ public void onConsumeResponse(BillingResult billingResult, String purchaseToken)
...
+                _dispatchEvent(ACKNOWLEDGE_ERROR, billingResult.getDebugMessage());
+            }
+        }
+    };
 
     private QueryInventoryFinishedListener _getProductsInfoListener = new QueryInventoryFinishedListener() {
         @Override
         public void onQueryInventoryFinished(Boolean success, String data) {
-
+
```
Comment (dmytrooverchenko) [inline android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/ExtensionContext.java]: Please remove this change.

File: android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/billingManager/BillingManagerV5.java [hunk 3]
```diff
@@ -125,17 +127,22 @@ public void onGetProductInfoFinishedListener(List<ProductDetails> productDetails
...
+                                            .build());
+                                            
+                                }
                             }
 
-                            QueryProductDetailsParams subParams = QueryProductDetailsParams.newBuilder()
-                                    .setProductList(subList)
-                                    .build();
+                            QueryProductDetailsParams subParams = null;
+                            if(subList.size()>0) {
```
Comment (dmytrooverchenko) [inline android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/billingManager/BillingManagerV5.java]: Please change to `if(subList.size() > 0)`

**Review by dmytrooverchenko — COMMENTED** [review]
Hello @mtilhan,
Thank you a lot for your contribution and sorry for the late reply. I left a few minor comments for the PR. Could you please fix them and I'll merge it right after that.

File: android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/ExtensionContext.java (hunk 2 above)
Comment (mtilhan) [reply android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/ExtensionContext.java]: Fixed

File: actionscript/src/com/freshplanet/ane/AirInAppPurchase/InAppPurchase.as (hunk 1 above)
Comment (mtilhan) [reply actionscript/src/com/freshplanet/ane/AirInAppPurchase/InAppPurchase.as]: Yes its better :) fixed

File: android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/billingManager/BillingManagerV5.java (hunk 3 above)
Comment (mtilhan) [reply android/lib/src/main/java/com/freshplanet/ane/AirInAppPurchase/billingManager/BillingManagerV5.java]: I see only space difference in the code here and I fixed it

**Review by dmytrooverchenko — APPROVED** [review]
Good job! Thank you

**Pull request merged**